aws/cf_template.go

```diff
@@ -227,7 +244,7 @@ func generateTemplate(spec *stackSpec) (string, error) {
 		resourceName := fmt.Sprintf("HTTPSListenerCertificate%x", hashARNs(certificateARNs))
 		template.AddResource(resourceName, &cloudformation.ElasticLoadBalancingV2ListenerCertificate{
 			Certificates: &certificateList,
-			ListenerArn:  cloudformation.Ref(listenerName).String(),
+			ListenerArn:  cloudformation.Ref("HTTPSListener").String(),
```

Comment:
I don't like the repetition of the inline string, maybe we can have a const for this.
We create a resource and we reference it with the same string, so maybe a const would be the right choice.

Reply:
well if we have
```go
const HTTPListener = "HTTPListener"
const HTTPSListener = "HTTPSListener"
```
it is as easy to mistype `HTTPListener vs HTTPSListener`  as `"HTTPListener"/"HTTPSListener"` but of course prevents `"HTTPXListener"`.

I have inlined
```go
httpProtocol  = "HTTP"
httpsProtocol = "HTTPS"
```
to remove the indirection - the `"HTTP"/"HTTPS"` are part of cloud formation DSL I do not see a reason why to put them into constant.

I am not strictly against constants here but I think we should tackle them systematically (needs thinking) and not only in a separate cases like this.

Reply:
I have no strong opinion on that, so let's go with this